File: java/src/tarehart/rlbot/input/BallTouch.java
package tarehart.rlbot.input;

import tarehart.rlbot.Bot;
import tarehart.rlbot.math.vector.Vector3;
import tarehart.rlbot.time.GameTime;

public class BallTouch {

    public final Bot.Team team;
    public final int playerIndex;
    public final GameTime time;
    public final Vector3 position;
    public final Vector3 normal;


    public BallTouch(Bot.Team team, int playerIndex, GameTime time, Vector3 position, Vector3 normal) {
        this.team = team;
        this.playerIndex = playerIndex;
        this.time = time;
        this.position = position;
        this.normal = normal;
    }
}

File: src/main/java/tarehart/rlbot/input/BallTouch.kt
package tarehart.rlbot.input

import tarehart.rlbot.bots.Team
import tarehart.rlbot.math.vector.Vector3
import tarehart.rlbot.time.GameTime

class BallTouch(val team: Team, val playerIndex: Int, val time: GameTime, val position: Vector3, val normal: Vector3)
